### PR TITLE
Added The Missing Boots Check

### PR DESCRIPTION
**Pull request by Thunder33345**
This is a sanity check to prevent unrelated non matching items get forcefully equipped as leggings
Item begin `Armor` does not also ensure it's an ID in the aforementioned arrays, assuming so THEN using boots as a "fallback" is a bad practice
And can lead to unintended consequences...

Example, what if i made my own plugin that register pumpkin as armor, now if i click with it in hand?
Your plugin would EXPECT `$item instanceof Armor` to means it MUST be vanilla implemented items,
That will be found in your predefined array, and pass it to `setArmorByType()`

It's all good and fine until now, but since pumpkin isnt in any of the arrays, and you used the boots as a fallback array
Now players are running around wearing pumpkins as boots!!!
Same thing can happen if i assume implement anything as armor, not saying you have the responsibility to handle equipping it,
But at the very least, your plugin should not to fall into an "undefined state"

TLDR: using "implied constraints"(instanceof armor, implies it is armor) in conjunction of using the last condition as "deduced match"(deducing, if it isnt helm, chest, leg, it MUST be boots) isnt a good idea

### CODE REVIEW

**Comment by Thunder33345** [comment]
Didnt see i accidentally left an indent, pls squash it or smth...

**Comment by RubyTemple** [comment]
thanks
